.github/workflows/ci.yml

```diff
@@ -35,12 +35,13 @@ jobs:
             ~/.cargo/bin
           key: ${{ runner.os }}-${{ env.MDBOOK_VERSION }}--${{ env.MDBOOK_LINKCHECK2_VERSION }}--${{ env.MDBOOK_TOC_VERSION }}--${{ env.MDBOOK_MERMAID_VERSION }}
 
-      - name: Cache linkcheck
-        uses: actions/cache@v4
```

Comment:
Any idea what this code was doing before? AFAICT it was doing nothing, which is pretty strange. It seems like it cached an empty linkcheck directory, which is useless, and also never restored it...

Reply:
The way the [cache](https://github.com/actions/cache) action works is that you use it at the beginning of the workflow, and then you generate the data that it should cache. The action then does this:
1) When the action is executed: if there is some cache entry, restore it from the cache. If there is no cache entry, do nothing.
2) At the end of the workflow: if the given path contains something, save it to a cache entry.

It's a little counter-intuitive in that you use it at the beginning, and it then uses an implicit step at the end of the workflow that actually writes to the cache.

Reply:
Thanks! I was wondering if it was something like that.